Tests/PointFreeTests/GhostTests.swift

```diff
@@ -0,0 +1,191 @@
+X-Content-Type-Options: nosniff
+X-Download-Options: noopen
+X-Frame-Options: SAMEORIGIN
+X-Permitted-Cross-Domain-Policies: none
+X-XSS-Protection: 1; mode=block
+""")
+  }
+
+  func testStartGhosting_NonAdmin() {
+    let adminUser = User.mock
+    let adminSession = Session.loggedIn
+      |> \.user .~ .standard(adminUser.id)
+
+    let ghostee = User.mock
+      |> \.id .~ User.Id(rawValue: UUID(uuidString: "10101010-dead-beef-dead-beefdeadbeef")!)
+
+    update(
+      &Current,
+      (\Environment.database) .~ .mock,
+      (\Environment.database.fetchUserById) .~ { userId -> EitherIO<Error, User?> in
+        pure(
+          userId == adminUser.id ? adminUser
+            : userId == ghostee.id ? ghostee
+            : nil
+        )
+      }
+    )
+
+    let conn = connection(
+      from: request(to: .admin(.ghost(.start(ghostee.id))), session: adminSession)
+      )
+      |> siteMiddleware
+      |> Prelude.perform
+
+    _assertInlineSnapshot(matching: conn, as: .conn, with: """
+POST http://localhost:8080/admin/ghost/start
+Cookie: pf_session={"userId":"00000000-0000-0000-0000-000000000000"}
+
+user_id=10101010-DEAD-BEEF-DEAD-BEEFDEADBEEF
+
+302 Found
+Location: /
+Referrer-Policy: strict-origin-when-cross-origin
+Set-Cookie: pf_session={"flash":{"message":"You don't have access to that.","priority":"error"},"userId":"00000000-0000-0000-0000-000000000000"}; Expires=Sat, 29 Jan 2028 00:00:00 GMT; Path=/
+X-Content-Type-Options: nosniff
+X-Download-Options: noopen
+X-Frame-Options: SAMEORIGIN
+X-Permitted-Cross-Domain-Policies: none
+X-XSS-Protection: 1; mode=block
+""")
+  }
+
+  func testEndGhosting_HappyPath() {
+    let ghostee = User.mock
+      |> \.id .~ User.Id(rawValue: UUID(uuidString: "10101010-dead-beef-dead-beefdeadbeef")!)
+
+    let adminUser = User.admin
+    let adminSession = Session.loggedIn
+      |> \.user .~ .ghosting(ghosteeId: ghostee.id, ghosterId: adminUser.id)
+
+    update(
+      &Current,
+      (\Environment.database) .~ .mock,
+      (\Environment.database.fetchUserById) .~ { userId -> EitherIO<Error, User?> in
+        pure(
+          userId == adminUser.id ? adminUser
+            : userId == ghostee.id ? ghostee
+            : nil
+        )
+      }
+    )
+
+    let conn = connection(
+      from: request(to: .endGhosting, session: adminSession)
+      )
+      |> siteMiddleware
+      |> Prelude.perform
+
+    _assertInlineSnapshot(matching: conn, as: .conn, with: """
+POST http://localhost:8080/ghosting/end
+Cookie: pf_session={"user":{"ghosteeId":"10101010-DEAD-BEEF-DEAD-BEEFDEADBEEF","ghosterId":"12121212-1212-1212-1212-121212121212"}}
+
+302 Found
+Location: /
+Referrer-Policy: strict-origin-when-cross-origin
+Set-Cookie: pf_session={"userId":"12121212-1212-1212-1212-121212121212"}; Expires=Sat, 29 Jan 2028 00:00:00 GMT; Path=/
```

Comment:
this test shows us transitioning back to our original account when ending ghosting.